src/js/components/Tabs/Tabs.js

```diff
@@ -93,37 +180,145 @@ const Tabs = forwardRef(
       messages,
     })}`;
 
+    const onLeft = () => {
+      if (focusIndex !== 0) {
+        tabRefs[focusIndex - 1].current.focus();
+        setFocusIndex(focusIndex - 1);
+      }
+    };
+
+    const onRight = () => {
+      if (focusIndex < tabRefs.length - 1) {
+        tabRefs[focusIndex + 1].current.focus();
+        setFocusIndex(focusIndex + 1);
+      }
+    };
+
+    const onHome = () => {
+      if (focusIndex !== 0 && tabRefs.length > 0) tabRefs[0].current.focus();
+    };
+
+    const onEnd = () => {
+      if (focusIndex !== 0 && tabRefs.length > 0)
+        tabRefs[tabRefs.length - 1].current.focus();
+    };
+
     return (
-      <StyledTabs
-        ref={ref}
-        as={Box}
-        role="tablist"
-        flex={flex}
-        responsive={responsive}
-        {...rest}
-        background={theme.tabs.background}
-      >
-        <StyledTabsHeader
+      <Keyboard onLeft={onLeft} onRight={onRight} onHome={onHome} onEnd={onEnd}>
+        <StyledTabs
+          ref={ref}
           as={Box}
-          direction="row"
-          justify={justify}
-          alignSelf={alignControls}
-          flex={false}
-          wrap
-          background={theme.tabs.header.background}
-          gap={theme.tabs.gap}
-          {...tabsHeaderStyles}
-        >
-          {tabs}
-        </StyledTabsHeader>
-        <StyledTabPanel
+          role="tablist"
           flex={flex}
-          aria-label={tabContentTitle}
-          role="tabpanel"
+          responsive={responsive}
+          {...rest}
+          background={theme.tabs.background}
         >
-          {activeContent}
-        </StyledTabPanel>
-      </StyledTabs>
+          <Box direction={overflow && scroll ? 'row' : 'column'}>
+            {overflow && scroll && (
+              <Button
+                a11yTitle="Previous Tab"
+                icon={<Previous />}
+                disabled={disableLeftArrow}
+                onClick={() => {
+                  let scrolledToIndex;
+                  for (let i = 0; i < tabRefs.length - 1; i += 1) {
+                    if (
+                      !isVisible(tabRefs[i].current) &&
+                      isVisible(tabRefs[i + 1].current)
+                    ) {
+                      if (scroll && scroll.interval) {
+                        i =
+                          i - (scroll.interval - 1) >= 0
+                            ? i - (scroll.interval - 1)
+                            : 0;
+                      }
+                      scrolledToIndex = i;
+                      tabRefs[i].current.scrollIntoView({ behavior: 'smooth' });
+                      break;
+                    }
+                  }
+
+                  setDisableRightArrow(false);
+                  if (scrolledToIndex === 0) {
+                    // wait for scroll animation to finish
```

Comment:
Do we need to let the animation finish? I think I expected it to become disabled more immediately

Reply:
If the button becomes disabled before the animation finishes, the animation will stop where it is and never continue. I'll try and mess around with this more though and see if there is a way around this because I agree, ideally the button could be disabled immediately and the animation would continue.

Reply:
I adjusted this code so that the button gets disabled as soon as the animation finishes instead of waiting half a second before being disabled. But I couldn't find a way around having to wait for the animation to finish.